File: src/main/java/com/fish/service/column/ColumnTopicService.java
package com.fish.service.column;

import com.fish.common.Result;
import com.fish.entity.pojo.ColumnTopic;
import com.fish.entity.vo.ColumnVO;
import com.mybatisflex.core.service.IService;

import java.util.ArrayList;

public interface ColumnTopicService extends IService<ColumnTopic> {
    Result<?> addRow(ColumnTopic columnTopic);
    Result<?> deleteRow(ColumnTopic columnTopic);
}

File: src/main/kotlin/com/fish/service/column/ColumnTopicService.kt
package com.fish.service.column

import com.fish.common.Result
import com.fish.entity.pojo.ColumnTopic
import com.fish.entity.vo.TopicVO
import com.mybatisflex.core.paginate.Page
import com.mybatisflex.core.service.IService

interface ColumnTopicService : IService<ColumnTopic> {
    fun getOptionalTopic(columnId: Long, pageNo: Int, pageSize: Int) : Result<Page<TopicVO>>
    fun addTopicToColumn(columnId: Long, topicIds: ArrayList<Long>) : Result<*>
    fun removeTopic(columnId: Long, topicIds: ArrayList<Long>) : Result<*>
}
